File: tlo_sort.py
#!/usr/bin/env python3

import io
import sys

iencoding = "utf-8"
oencoding = "utf-8"
input_stream = io.TextIOWrapper(sys.stdin.buffer, encoding=iencoding)
output_stream = io.TextIOWrapper(sys.stdout.buffer, encoding=oencoding)

lines = []

for line in input_stream:
    lines.append(line)

lines.sort()

for line in lines:
    print(line, file=output_stream, end='')

File: tlo-sort.rb
#!/usr/bin/env ruby

require "fcntl"

iencoding = "utf-8:utf-8"
oencoding = "utf-8:utf-8"
input_stream = IO.new(STDIN.fcntl(Fcntl::F_DUPFD), encoding: iencoding)
output_stream = IO.new(STDOUT.fcntl(Fcntl::F_DUPFD), encoding: oencoding)

lines = []

input_stream.each do |line|
  lines.push(line)
end

lines.sort!

lines.each do |line|
  print(line)
end
